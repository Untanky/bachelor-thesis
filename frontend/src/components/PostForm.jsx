import React from 'react';
import PropTypes from 'prop-types';

import './PostForm.scss';
import TextInput from './TextInput';

import './PostForm.scss';

const PostForm = ({ initialState, onChange }) => {
  const onChangeField = (key) => (value) => {
    onChange(key, value);
  }

  return (
    <form className="post-form">
      <TextInput
        label="Title"
        onChange={onChangeField('title')}
        placeholder="Title"
        initialValue={initialState.title}
      />
      <TextInput 
        label="Description"
        onChange={onChangeField('description')}
        placeholder="Description"
        initialValue={initialState.description}
      />
    </form>
  )
};

PostForm.propTypes = {
  initialState: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
  }),
  onChange: PropTypes.func,
};

PostForm.defaultProps = {
  initialState: {
    title: '',
    description: '',
  },
};

export default PostForm;
