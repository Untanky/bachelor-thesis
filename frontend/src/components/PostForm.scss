.post-form {
  .text-input {
    margin: 0.5em 0;
  }
}