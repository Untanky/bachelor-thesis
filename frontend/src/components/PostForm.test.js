import React from 'react';
import PostForm from './PostForm';
import renderer from 'react-test-renderer';

global.Math.random = () => 0.5;

it('renders correctly with initial state', () => {
  const initialState = {
    title: 'Test title',
    description: 'Description might be longer than the title and contain more data',
  };

  const tree = renderer
    .create(<PostForm initialState={initialState} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without initial state', () => {
  const tree = renderer
    .create(<PostForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
