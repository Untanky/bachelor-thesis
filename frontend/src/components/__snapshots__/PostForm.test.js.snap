// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with initial state 1`] = `
<form
  className="post-form"
>
  <div
    className="text-input"
  >
    <label
      className="label"
      htmlFor={0.5}
    >
      Title
    </label>
    <input
      className="input-field"
      id={0.5}
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value="Test title"
    />
  </div>
  <div
    className="text-input"
  >
    <label
      className="label"
      htmlFor={0.5}
    >
      Description
    </label>
    <input
      className="input-field"
      id={0.5}
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value="Description might be longer than the title and contain more data"
    />
  </div>
</form>
`;

exports[`renders correctly without initial state 1`] = `
<form
  className="post-form"
>
  <div
    className="text-input"
  >
    <label
      className="label"
      htmlFor={0.5}
    >
      Title
    </label>
    <input
      className="input-field"
      id={0.5}
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value=""
    />
  </div>
  <div
    className="text-input"
  >
    <label
      className="label"
      htmlFor={0.5}
    >
      Description
    </label>
    <input
      className="input-field"
      id={0.5}
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
  </div>
</form>
`;
